tycho-sbom/src/main/java/org/eclipse/tycho/sbom/TychoProjectDependenciesConverter.java

```diff
@@ -108,38 +111,39 @@ private void convertToDependency(DependencyTreeNode node, Set<Dependency> depend
 	 * empty list is returned.
 	 * 
 	 * @param iu The installable unit for which to generate
-	 * @return A {@code mutable} list of all bom representation. matching the given
-	 *         IU.
+	 * @return An {@code immutable} list of all bom representations matching the
+	 *         given IU.
 	 */
 	private List<String> getBomRepresentation(IInstallableUnit iu) {
-		final MavenSession mavenSession = legacySupport.getSession();
-		final List<MavenProject> reactorProjects = mavenSession.getAllProjects();
-		// mutable!
-		List<String> bomRefs = new ArrayList<>();
-		// (I) IU describes local reactor project
-		for (MavenProject project : reactorProjects) {
-			ReactorProject reactorProject = DefaultReactorProject.adapt(project);
-			Set<IInstallableUnit> initalUnits = reactorProject.getDependencyMetadata(DependencyMetadataType.INITIAL);
-			Set<IInstallableUnit> seedUnits = reactorProject.getDependencyMetadata(DependencyMetadataType.SEED);
-			if (initalUnits.contains(iu) || seedUnits.contains(iu)) {
-				String bomRef = modelConverter.generatePackageUrl(project.getArtifact());
+		return bomRepresentations.computeIfAbsent(iu, ignore -> {
+			final MavenSession mavenSession = legacySupport.getSession();
+			final List<MavenProject> reactorProjects = mavenSession.getAllProjects();
+			final List<String> bomRefs = new ArrayList<>();
+			// (I) IU describes local reactor project
+			for (MavenProject project : reactorProjects) {
+				ReactorProject reactorProject = DefaultReactorProject.adapt(project);
+				Set<IInstallableUnit> initalUnits = reactorProject.getDependencyMetadata(DependencyMetadataType.INITIAL);
+				Set<IInstallableUnit> seedUnits = reactorProject.getDependencyMetadata(DependencyMetadataType.SEED);
+				if (initalUnits.contains(iu) || seedUnits.contains(iu)) {
+					String bomRef = modelConverter.generatePackageUrl(project.getArtifact());
+					if (bomRef == null) {
+						LOG.error("Unable to calculate BOM for: " + project);
+						return List.copyOf(bomRefs);
+					}
+					bomRefs.add(bomRef);
+					return List.copyOf(bomRefs);
```

Comment:
```suggestion
					return List.of(bomRef);
```

Reply:
This won't compile as `bomRefs` is already a list. But to avoid copying the array, I now use `Collections.unmodifiableList(...)`